### DISCUSSION TITLE
Trigger clip if not playing

### DISCUSSION THREAD

**Discussion by bendorge** [leolabs/ableton-js#116] [Q&A] [answered]
Using AbletonJS, what code would be required to check that a clipslot in a particular scene is not playing, and then play that clip?

**Comment by leolabs** [comment, marked as answer]
Hey @bendorge,

Here's a simple example that starts the clip in the first slot of the first track if it's not playing:

```ts
const ableton = new Ableton();
await ableton.start();

const tracks = await ableton.song.get("tracks");
const clipSlots = await tracks[0].get("clip_slots");

const firstSlot = clipSlots[0];
const isPlaying = await firstSlot.get("is_playing");

if (!isPlaying) {
  await firstSlot.fire();
} else {
  await firstSlot.stop();
}
```

Let me know if this helps!

**Reply by bendorge** [reply]
This is working for sure. Thank you so much! I'll have to experiment a little with clip toggling via JavaScript. The initial fire() works great, but toggling off using .fire() doesn't seem to work. One thing I thought I would share is that if you have multiple tracks with a clipslot to fire, removing "await" plays them all at once, which is awesome. I'll have to dig through that API a little bit more...

**Reply by leolabs** [reply]
I'm glad I could help! If you want to stop a clip, you can call `stop()` on the ClipSlot, that should do the trick. I've updated the answer with an example.

Leaving out the `await` can be good for executing multiple commands at once, just be aware that any errors that might occur during those calls won't be caught and just printed to the console by default.

I'd use `Promise.all` to run multiple commands in parallel:

```ts
const tracks = await ableton.song.get("tracks");

// This toggles the first clip of each track in the set if it has one
await Promise.all(tracks.map(async track => {
  const clipSlots = await track.get("clip_slots");
  const isPlaying = await clipSlots[0].get("is_playing");

  if (!isPlaying) {
    await clipSlots[0].fire();
  } else {
    await clipSlots[0].stop();
  }
}));
```

**Reply by bendorge** [reply]
Yep, that worked as well! Thanks again for the updated code!
